Packs/Tenable_io/Integrations/TenableioEventCollector/TenableioEventCollector.py

```diff
@@ -0,0 +1,446 @@
+        return status, chunks_available
+
+    def download_vulnerabilities_chunk(self, export_uuid: str, chunk_id: int):
+        """
+
+        Args:
+            export_uuid: The UUID of the vulnerabilities export job.
+            chunk_id: The ID of the chunk you want to export.
+
+        Returns: Chunk of vulnerabilities from API.
+
+        """
+        return self._http_request(method='GET', url_suffix=f'/vulns/export/{export_uuid}/chunks/{chunk_id}',
+                                  headers=self._headers)
+
+
+''' HELPER FUNCTIONS '''
+
+
+def try_get_chunks(client: Client, export_uuid: str):
+    """
+    If job has succeeded (status FINISHED) get all information from all chunks available.
+    Args:
+        client: Client class object.
+        export_uuid: The UUID of the vulnerabilities export job.
+
+    Returns: All information from all chunks available.
+
+    """
+    vulnerabilities = []
+    status, chunks_available = client.get_export_status(export_uuid=export_uuid)
+    demisto.info(f'Report status is {status}, and number of available chunks is {chunks_available}')
+    if status == 'FINISHED':
+        for chunk_id in chunks_available:
+            vulnerabilities.extend(client.download_vulnerabilities_chunk(export_uuid=export_uuid, chunk_id=chunk_id))
+    return vulnerabilities, status
+
+
+def generate_export_uuid(client: Client, first_fetch: Optional[datetime], last_run: Dict[str, str | float | None],
+                         severity: List[str]):
+    """
+    Generate a job export uuid in order to fetch vulnerabilities.
+
+    Args:
+        client: Client class object.
+        first_fetch: time to first fetch from.
+        last_run: last run object.
+        severity: severity of the vulnerabilities to include in the export.
+
+    """
+    demisto.info("Getting export uuid for report.")
+    last_fetch = last_run.get('last_fetch_vuln')
+    last_found: float = time.mktime(
+        first_fetch.timetuple()) if not last_fetch and first_fetch else last_fetch  # type: ignore
+    export_uuid = client.get_export_uuid(num_assets=5000, last_found=last_found, severity=severity)
+
+    next_run_vuln = time.mktime(datetime.now(tz=timezone.utc).timetuple())
+    demisto.info(f'export uuid is {export_uuid}')
+    last_run.update({'last_found_use': last_found, 'last_fetch_vuln': next_run_vuln, 'export_uuid': export_uuid})
+
+
+def run_vulnerabilities_fetch(last_run: dict, first_fetch: Optional[datetime], vuln_fetch_interval: int):
+    """
+
+    Args:
+        last_run: last run object.
+        first_fetch: time to first fetch from.
+        vuln_fetch_interval: vulnerabilities fetch interval.
+
+    Returns: True if fetch vulnerabilities can be run.
+
+    """
+    if not last_run.get('last_fetch_vuln'):
+        time_to_check = time.mktime(first_fetch.timetuple())  # type: ignore
+    else:
+        time_to_check = last_run.get('last_fetch_vuln')  # type: ignore
```

Comment:
general note: try using typing instead of `type: ignore` (it's almost always possible to avoid this)